engine/runtime/src/main/java/org/enso/interpreter/runtime/data/hash/EnsoHashMap.java

```diff
@@ -165,14 +165,15 @@ Type getMetaObject(@CachedLibrary("this") InteropLibrary thisLib) {
 
   @ExportMessage
   @TruffleBoundary
-  Object toDisplayString(boolean allowSideEffects) {
+  Object toDisplayString(
+      boolean allowSideEffects, @CachedLibrary(limit = "5") InteropLibrary interop) {
```

Comment:
Don't use `@CachedLibrary` for `@TruffleBoundary` message - use `InteropLibrary.getUncached()`.

Reply:
Good point. Specializations behind `@TruffleBoundary` initialize all the `@Cached` and `@CachedLibrary` annotated parameters with their uncached variants anyway.

Reply:
> Specializations behind @TruffleBoundary initialize all the @Cached and @CachedLibrary annotated parameters with their uncached variants anyway.

I don't think they do. Looking at `EnsoHashMapGen.java` I see:
```java
            private Object toDisplayStringNode_AndSpecialize(EnsoHashMap arg0Value, boolean arg1Value) {
                Lock lock = getLock();
                boolean hasLock = true;
                lock.lock();
                try {
                    int state_0 = this.state_0_;
                    this.toDisplayStringNode__toDisplayString_interop_ = super.insert((INTEROP_LIBRARY_.createDispatched(5)));
                    this.state_0_ = state_0 = state_0 | 0b10 /* add-state_0 toDisplayString(EnsoHashMap, boolean, InteropLibrary) */;
                    lock.unlock();
                    hasLock = false;
                    return arg0Value.toDisplayString(arg1Value, this.toDisplayStringNode__toDisplayString_interop_);
```
which seems to call `createDispatched`.

Reply:
Please remove the `InteropLibrary interop` parameter from the `toDisplayString` method. Instead use `var interop = InteropLibrary.getUncached()` inside of the method.

Reply:
Done in my upcomming PR - cba65de7c8d70a73126f6bc0126a8c9b7d7d33b8